activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/main/java/org/activiti/cloud/services/rest/controllers/CandidateAdminControllerImpl.java

```diff
@@ -0,0 +1,110 @@
+package org.activiti.cloud.services.rest.controllers;
+
+import org.activiti.api.task.model.payloads.CandidateGroupsPayload;
+import org.activiti.api.task.model.payloads.CandidateUsersPayload;
+import org.activiti.api.task.runtime.TaskAdminRuntime;
+import org.activiti.cloud.api.process.model.impl.CandidateGroup;
+import org.activiti.cloud.api.process.model.impl.CandidateUser;
+import org.activiti.cloud.services.rest.api.CandidateAdminController;
+import org.activiti.cloud.services.rest.assemblers.GroupCandidatesResourceAssembler;
+import org.activiti.cloud.services.rest.assemblers.ResourcesAssembler;
+import org.activiti.cloud.services.rest.assemblers.ToCandidateGroupConverter;
+import org.activiti.cloud.services.rest.assemblers.ToCandidateUserConverter;
+import org.activiti.cloud.services.rest.assemblers.UserCandidatesResourceAssembler;
+import org.springframework.hateoas.Resource;
+import org.springframework.hateoas.Resources;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.methodOn;
+
+@RestController
+public class CandidateAdminControllerImpl implements CandidateAdminController {
+
+    private final TaskAdminRuntime taskAdminRuntime;
+
+    private final UserCandidatesResourceAssembler userCandidatesResourceAssembler;
+
+    private final GroupCandidatesResourceAssembler groupCandidatesResourceAssembler;
+
+    private final ToCandidateUserConverter toCandidateUserConverter;
+
+    private final ToCandidateGroupConverter toCandidateGroupConverter;
+
+    private final ResourcesAssembler resourcesAssembler;
+
+    public CandidateAdminControllerImpl(TaskAdminRuntime taskAdminRuntime,
```

Comment:
I think you could go even further with controller split and have CandidateUserAdminController and CandiateGroupAdminController. This would reduce even more the number of parameters and you could add `/v1/tasks/{taskId}/candidate-users` and `/v1/tasks/{taskId}/candiate-groups` directly in the interface level.